### PR TITLE
feat: Make LoRA alpha and dropout configurable, adjust SFT training s…

### PR DESCRIPTION
**Pull request by droot**
…teps, limits, and learning rate, and update test server dependencies.

### CODE REVIEW

**Comment by droot** [comment]
@ShubyM I am merging this. I have tested this for pig-latin fine-tune of gemma-3-1b and qwen. I can address feedback (if any) in the follow ups.
